lib/puppet/indirector/node/write_only_yaml.rb

```diff
@@ -11,7 +11,12 @@
 # @api private
 #
 class Puppet::Node::WriteOnlyYaml < Puppet::Indirector::Yaml
-  desc "Store node information as flat files, serialized using YAML,
+  def initialize
+    Puppet.deprecation_warning('Puppet::Node::WriteOnlyYaml is deprecated and will be removed in a future release of Puppet.')
+    super
+  end
+
+  desc "(Deprecated) Store node information as flat files, serialized using YAML,
```

Comment:
I think so. My thinking was a) we're trying to move away from yaml serializing to disk (well really yaml entirely) and b) there are no users of this terminus in the codebase after this PR so why ship it?